tests/workflow_task_reported_problems_test.go

```diff
@@ -0,0 +1,360 @@
+		TaskQueue: s.TaskQueue(),
+	}
+
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, s.simpleWorkflow)
+	s.NoError(err)
+
+	// Make sure the workflow has started and had an activity task scheduled and finished
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.Equal(t, enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING, description.WorkflowExecutionInfo.Status)
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Check if the search attributes are not empty and has TemporalReportedProblems
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		require.NoError(t, err)
+		require.NotNil(t, description.WorkflowExecutionInfo.SearchAttributes)
+		require.NotEmpty(t, description.WorkflowExecutionInfo.SearchAttributes.IndexedFields)
+		require.NotNil(t, description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems])
+
+		// Decode the search attribute in keyword list format
+		searchValBytes := description.WorkflowExecutionInfo.SearchAttributes.IndexedFields[searchattribute.TemporalReportedProblems]
+		searchVal, err := searchattribute.DecodeValue(searchValBytes, enumspb.INDEXED_VALUE_TYPE_KEYWORD_LIST, false)
+		require.NoError(t, err)
+		require.NotEmpty(t, searchVal)
+		require.Equal(t, "category=WorkflowTaskFailed", searchVal.([]string)[0])
+		require.Equal(t, "cause=WorkflowWorkerUnhandledFailure", searchVal.([]string)[1])
+
+		// Validate attempt number after verifying search attribute values
+		require.GreaterOrEqual(t, description.GetPendingWorkflowTask().Attempt, int32(2))
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Check if the search attributes are queryable
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		queriedWorkflows, err := s.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			Query:     "TemporalReportedProblems IS NOT NULL",
+			PageSize:  100,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(queriedWorkflows.Executions))
+
+		queriedWorkflows, err = s.SdkClient().ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: s.Namespace().String(),
+			Query:     "TemporalReportedProblems IN ('category=WorkflowTaskFailed', 'cause=WorkflowWorkerUnhandledFailure')",
+			PageSize:  100,
+		})
+		require.NoError(t, err)
+		require.Equal(t, 1, len(queriedWorkflows.Executions))
+	}, 5*time.Second, 500*time.Millisecond)
+
+	// Unblock the workflow
+	s.shouldFail.Store(false)
+
+	var out string
+	err = workflowRun.Get(ctx, &out)
+
+	s.NoError(err)
+
+	// Validate the workflow completed successfully and the search attribute is removed
+	s.EventuallyWithT(func(t *assert.CollectT) {
```

Comment:
No need to use `Eventually` here, describe is a consistent API and you've already waited for the workflow to complete above.